cephadm-preflight.yml

```
@@ -47,278 +47,15 @@
-                    state: "{{ custom_repo_state | default('present') }}"
-                    gpgkey: "{{ custom_repo_gpgkey | default(omit) }}"
-                    baseurl: "{{ custom_repo_url }}"
-                    enabled: "{{ custom_repo_enabled | default(1) }}"
-                    file: ceph_custom
-                    priority: '2'
-              when: ceph_custom_repositories is undefined
-
-            - name: setup custom repositories
-              yum_repository:
-                name: "{{ item.name }}"
-                description: "{{ item.description }}"
-                state: "{{ item.state | default(omit) }}"
-                gpgcheck: "{{ item.gpgcheck | default(omit) }}"
-                gpgkey: "{{ item.gpgkey | default(omit) }}"
-                baseurl: "{{ item.baseurl }}"
-                file: "{{ item.ceph_custom | default(omit) }}"
-                priority: "{{ item.priority | default(omit) }}"
-                enabled: "{{ item.enabled | default(omit) }}"
-              register: result
-              until: result is succeeded
-              loop: "{{ ceph_custom_repositories }}"
-
-        - name: install epel-release
-          when: ansible_facts['distribution'] != 'RedHat'
-          block:
-            - name: enable required CentOS repository for epel
-              command: dnf config-manager --set-enabled "{{ 'powertools' if ansible_facts['distribution_major_version'] == '8' else 'crb' }}"
-              changed_when: false
-
-            - name: install package
-              package:
-                name: epel-release
-                state: present
-              register: result
-              until: result is succeeded
-
-        - name: remove remaining local services ceph packages
-          dnf:
-            name: "{{ packages_to_uninstall }}"
-            state: absent
-            autoremove: false
-
-        - name: install ceph-common on rhel
-          package:
-            name: ceph-common
-            state: "{{ (upgrade_ceph_packages | bool) | ternary('latest', 'present') }}"
-          register: result
-          until: result is succeeded
-
-        - name: install prerequisites packages on servers
-          package:
-            name: "{{ ceph_pkgs + infra_pkgs }}"
-            state: "{{ (upgrade_ceph_packages | bool) | ternary('latest', 'present') }}"
-          register: result
-          until: result is succeeded
-          when: group_names != [client_group]
-
-        - name: install prerequisites packages on clients
-          package:
-            name: "{{ ceph_client_pkgs }}"
-            state: "{{ (upgrade_ceph_packages | bool) | ternary('latest', 'present') }}"
-          register: result
-          until: result is succeeded
-          when: group_names == [client_group]
-
-
-        - name: ensure chronyd is running
-          service:
-            name: chronyd
-            state: started
-            enabled: true
+      import_tasks: tasks/redhat.yml
```

just a nit, another approach:

```suggestion
      - name: Import specific OS-related tasks
        import_tasks: "{{ item }}"
        with_first_found:
          - "{{ ansible_distribution }}.yml"
```

with:

```
tasks/
├── Ubuntu.yml
├── Debian.yml
├── CentOS.yml -> RedHat.yml
└── RedHat.yml
```